runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -862,20 +892,36 @@ void chpl_comm_init(int *argc_p, char ***argv_p) {
   set_max_segsize();
   set_num_comm_domains();
   setup_ibv();
-  setup_polling();
 
-  GASNET_Safe(gex_Client_Init(&myclient, &myep, &myteam, "chapel", argc_p, argv_p, GEX_FLAG_USES_GASNET1));
+  usePSHM = chpl_env_rt_get_bool("COMM_GASNET_USE_PSHM", usePSHM);
+  if (!usePSHM) {
+    // Setting this to 1 will disable PSHM even if it was enabled during
+    // GASNet configuration.
+    chpl_env_set("GASNET_SUPERNODE_MAXSIZE", "1", 0);
```

Comment:
On a related note, it seems problematic to introduce a `CHPL_COMM_GASNET_USE_PSHM` envvar that has overlapping behavior with `GASNET_SUPERNODE_MAXSIZE`. So for example if a user sets: 

1. Use case 1: `CHPL_COMM_GASNET_USE_PSHM=1 GASNET_SUPERNODE_MAXSIZE=1` or
2. Use case 2: `CHPL_COMM_GASNET_USE_PSHM=0 GASNET_SUPERNODE_MAXSIZE=0`, 

It's very non-obvious that these each represent conflicting inputs, and that the behavior will depend critically on the details of this conflict is handled by the backend.